src/commands/stepfunctions/awsCommands.ts

```diff
@@ -205,7 +205,13 @@ export const attachPolicyToStateMachineIamRole = async (
   context: BaseContext,
   dryRun: boolean
 ): Promise<AttachRolePolicyCommandOutput | undefined> => {
-  const roleName = describeStateMachineCommandOutput?.roleArn?.split('/')[1]
+  const splitRoleArnList = describeStateMachineCommandOutput?.roleArn?.split('/')
+  if (splitRoleArnList === undefined) {
+    throw Error(
+      `unexpected roleArn ${describeStateMachineCommandOutput?.roleArn} for the describeStateMachineCommandOutput ${describeStateMachineCommandOutput}`
+    )
+  }
+  const roleName = splitRoleArnList[splitRoleArnList.length - 1]
```

Comment:
Should we add some comment explaining the different name format between standard and express state machines,  maybe with an example, and say that this code covers both cases? Thus future code maintainers can be aware of this caveat.